### PR TITLE
Adding script to generate cgmanifest.json for transitive dependencies

### PR DESCRIPTION
**Pull request by conniey**
Our current tooling for searching for dependencies only examines immediate dependencies. It's possible for dependency `Foo` to rely on a version of `Bar` that requires an update. Our tooling doesn't pick up dependency `Bar`.

* This script generates a `cgmanifest.json`, or appends to an existing one. 
* Sample: `pwsh .\eng\Set-ComponentRegistrations.ps1 D:\git\conniey\azure-sdk-for-java-2\sdk\eventhubs\ -ExcludeRegex "mgmt-*"`
  * Will search under _eventhubs_ for pom.xml and will exclude any directories that start with "mgmt-*".

/cc @JimSuplizio @joshfree

### CODE REVIEW

File: sdk/servicebus/microsoft-azure-servicebus/cgmanifest.json [hunk 1]
```diff
@@ -0,0 +1,475 @@
+{
```
Comment (JimSuplizio) [inline sdk/servicebus/microsoft-azure-servicebus/cgmanifest.json]: @conniey If we're generating a manifest as part of the build step should this cgmanifest file be checked in?

Comment (conniey) [reply sdk/servicebus/microsoft-azure-servicebus/cgmanifest.json]: Yeah, this should be removed. I'll delete it.

**Comment by mitchdenny** [comment]
Feels like we are doing a lot of heavy lifting here for something that component governance should be doing. Did you speak to anyone on the component governance team about this already?

**Comment by conniey** [comment]
@mitchdenny Hey. Yes, I did. They said it's not a scenario they support and to create a cgmanifest.json for these transitive dependencies.

**Comment by mitchdenny** [comment]
> @mitchdenny Hey. Yes, I did. They said it's not a scenario they support and to create a cgmanifest.json for these transitive dependencies.

Interesting. If it doesn't matter to them, why does it matter to us?

**Comment by conniey** [comment]
I tried to explain it in the summary, but concrete examples are here, where the dependency itself does not contain any issues, but the dependencies it pulls in does. Tools like [DependencyCheck](https://owasp.org/www-project-dependency-check/) do check for these transitive dependencies.
* https://github.com/Azure/azure-sdk-for-java/issues/8440
* https://github.com/Azure/azure-sdk-for-java/pull/8247

**Pull request merged**